apps/cowswap-frontend/src/modules/trade/utils/swapErrorHelper.ts

```diff
@@ -1,11 +1,25 @@
 import { getProviderErrorMessage, isRejectRequestProviderError } from '@cowprotocol/common-utils'
 
+import { getIsOrderBookTypedError } from 'api/gnosisProtocol'
+
 export const USER_SWAP_REJECTED_ERROR = 'User rejected signing the order'
 
 export function getSwapErrorMessage(error: Error): string {
   if (isRejectRequestProviderError(error)) {
     return USER_SWAP_REJECTED_ERROR
   } else {
-    return getProviderErrorMessage(error)
+    const defaultErrorMessage = getProviderErrorMessage(error)
+
+    if (getIsOrderBookTypedError(error)) {
+      return capitalizeFirstLetter(error.body?.description) || defaultErrorMessage
```

Comment:
don't we do the mapping if the type is know? 

https://github.com/cowprotocol/cowswap/blob/679e9fafa4c119c95f5bb531378aa7c32358e307/apps/cowswap-frontend/src/api/gnosisProtocol/errors/OperatorError.ts#L44

Reply:
I think it's better to show an error as it is. Having this we can change the text on backend side on flight